### PR TITLE
Add Jetpack and Layout Grid translations

### LINKED ISSUE #2615: Strings from Jetpack are not included in the bundle
**Describe the bug**
When we run our bundle script we are not including strings from jetpack in the resulting bundle.

**To Reproduce**
1. `npm run genstrings`
2. Observe that the strings files (i.e., `bundle/android/strings.xml`) do not contain relevant strings from the jetpack repo: for example, ["Link address to Google Maps"](https://github.com/Automattic/jetpack/blob/f6f313d268c336491c8bca1afcf3e5d3663597ce/extensions/blocks/contact-info/address/edit.native.js#L42)). 

**Expected behavior**
Relevant strings from the jetpack repos should be included in our bundles for translation.

### PR DESCRIPTION
**Pull request by fluiddot**
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/2615

## Description

This PR adds the translation files for localizing Jetpack blocks and the Layout Grid block (block experiments).

**I'd like to note that, although this PR adds the missing translations, it's only the first part of the process of the incorporation process, as these changes don't bring the logic for updating them.**

### Translation files
For this purpose, a new folder located at `src/i18n-translations` has been created and contains the translations for each project. These files have been generated as part of a different PR (https://github.com/wordpress-mobile/gutenberg-mobile/pull/4366), where optimized translation files will be produced including only the strings used by the native version of the editor.

**NOTE:** In the future, this folder won't be necessary as translations will be generated on the fly via a new command and when producing the JS bundle.

### POT files
For the strings that are not part of the web version and therefore not included in the plugin's GlotPress project, we have to include them in the localization strings files (`bundle/ios/GutenbergNativeTranslations.swift` / `bundle/android/strings.xml`), exactly the same as we do for the Gutenberg strings.

This implied adding new NPM commands to generate the required POT files and updating the scripts we use for generating the localization strings files (`po2swift.js` / `po2android.js`), so they support multiple POT files as now we'll have one file each Gutenberg, Jetpack, and Layout Grid projects.

**NOTE:** I'm working on a new command to unify the POT files generation into a single one, so eventually, we won't need to add extra commands when adding translations of new plugins. 

### Localization strings files

The strings added to `bundle/ios/GutenbergNativeTranslations.swift` and `bundle/android/strings.xml` files are strings coming from Jetpack and Layout Grid plugins that require translation as they're only referenced in `.native.js` files.

### Verify that localization strings files are not modified
- Run command: `npm run genstrings`
- Observe that neither `bundle/ios/GutenbergNativeTranslations.swift` and `bundle/android/strings.xml` are modified.

### Verify that Jetpack translations are present
1. Change the device's language to non-English.
1. Open the app.
1. Insert a Jetpack block (i.e.: Contact Info or Story blocks)
1. Observe that strings related to the block are translated to the selected language.
1. If any string remains untranslated, verify that is included in the localization strings files (`bundle/ios/GutenbergNativeTranslations.swift` / `bundle/android/strings.xml`). 

### Verify that Layout Grid translations are present
1. Change the device's language to non-English.
1. Open the app.
1. Insert the Layout Grid block
1. Observe that strings related to the block are translated to the selected language.
1. If any string remains untranslated, verify that is included in the localization strings files (`bundle/ios/GutenbergNativeTranslations.swift` / `bundle/android/strings.xml`).

**NOTE:** I noticed that some strings of the Layout Grid block are not translated, looks like this is caused because they're not retrieved by the correct domain ([reference](https://github.com/Automattic/block-experiments/blob/00dbb963f1615d9843f0d8a8479b585c7a0108e1/blocks/layout-grid/src/grid/variations.js#L20)) hence, they're not being included in [the GlotPress project](https://translate.wordpress.org/projects/wp-plugins/layout-grid/). We'll address this in a separate PR.

<details><summary>Click here to display screenshot</summary>

</details>

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -74,15 +74,21 @@
...
+		"makepot:android:layout-grid": "npm run wp -- i18n make-pot ./block-experiments/blocks/layout-grid --include=*.native.js,*.android.js --exclude=test/*,e2e-tests/*,build/*,build-module/*,build-style/* --skip-php --subtract=./layout-grid.pot --ignore-domain layout-grid-android.pot",
+		"makepot:ios:gutenberg": "npm run wp -- i18n make-pot ./gutenberg/packages --include=*.native.js,*.ios.js --exclude=test/*,e2e-tests/*,build/*,build-module/*,build-style/* --skip-php --subtract=./gutenberg.pot --ignore-domain gutenberg-ios.pot",
+		"makepot:ios:jetpack": "npm run wp -- i18n make-pot ./jetpack/projects/plugins/jetpack/extensions --include=*.native.js,*.ios.js --exclude=test/*,e2e-tests/*,build/*,build-module/*,build-style/* --skip-php --subtract=./jetpack.pot --ignore-domain jetpack-ios.pot",
+		"makepot:ios:layout-grid": "npm run wp -- i18n make-pot ./block-experiments/blocks/layout-grid --include=*.native.js,*.ios.js --exclude=test/*,e2e-tests/*,build/*,build-module/*,build-style/* --skip-php --subtract=./layout-grid.pot --ignore-domain layout-grid-ios.pot",
```
Comment (fluiddot) [inline package.json:82]: Generate POT files for each plugin including Gutenberg. This approach won't scale easily as we add more plugins, so I'll provide in a new PR a way to unify the POT files creation in a single command.

File: package.json [hunk 2]
```diff
@@ -74,15 +74,21 @@
...
 		"premakepot:gutenberg": "npm run clean:gutenberg && npm run build:gutenberg",
 		"makepot:gutenberg": "npm run wp -- i18n make-pot ./gutenberg/packages --exclude=test/*,e2e-tests/*,build/*,build-module/*,build-style/*,*.native.js,*.ios.js,*.android.js,bundle/* --ignore-domain gutenberg.pot",
 		"postmakepot:gutenberg": "npm run clean:gutenberg",
-		"pregenstrings": "npm run makepot:gutenberg",
+		"makepot:jetpack": "npm run wp -- i18n make-pot ./jetpack/projects/plugins/jetpack/extensions --exclude=test/*,e2e-tests/*,build/*,build-module/*,build-style/*,*.native.js,*.ios.js,*.android.js,bundle/* --ignore-domain jetpack.pot",
+		"makepot:layout-grid": "npm run wp -- i18n make-pot ./block-experiments/blocks/layout-grid --exclude=test/*,e2e-tests/*,build/*,build-module/*,build-style/*,*.native.js,*.ios.js,*.android.js,bundle/* --ignore-domain layout-grid.pot",
```
Comment (fluiddot) [inline package.json:87]: Similar to Gutenberg, we need to generate these POT files for each plugin to subtract strings on `makepot:<platform>:<plugin>` above commands.

File: package.json [hunk 3]
```diff
@@ -74,15 +74,21 @@
...
+		"pregenstrings": "npm run makepot:gutenberg && npm run makepot:jetpack && npm run makepot:layout-grid",
 		"genstrings": "npm run genstrings:android && npm run genstrings:ios",
-		"genstrings:android": "npm run makepot:android && ./bin/po2android.js gutenberg-android.pot bundle/android/strings.xml",
-		"genstrings:ios": "npm run makepot:ios && ./bin/po2swift.js gutenberg-ios.pot bundle/ios/GutenbergNativeTranslations.swift",
+		"genstrings:android": "npm run makepot:android:gutenberg && npm run makepot:android:jetpack && npm run makepot:android:layout-grid && ./bin/po2android.js bundle/android/strings.xml gutenberg-android.pot jetpack-android.pot layout-grid-android.pot",
+		"genstrings:ios": "npm run makepot:ios:gutenberg && npm run makepot:ios:jetpack && npm run makepot:ios:layout-grid && ./bin/po2swift.js bundle/ios/GutenbergNativeTranslations.swift gutenberg-ios.pot jetpack-ios.pot layout-grid-ios.pot",
```
Comment (fluiddot) [inline package.json:91]: As we do for Gutenberg, we have also to include the POT files generated for each plugin (Jetpack + Layout Grid) when generating the localization strings files.

File: bin/po2android.js [hunk 4]
```diff
@@ -117,10 +120,10 @@ ${ indent }</string-array>
...
 if ( require.main === module ) {
 	if ( process.stdin.isTTY ) {
-		const potFileName = process.argv[ 2 ];
-		const destination = process.argv[ 3 ];
-		const potFileContent = fs.readFileSync( potFileName );
-		const xmlOutput = po2Android( potFileContent, process.argv[ 3 ] );
+		const destination = process.argv[ 2 ];
+		const potFiles = process.argv.slice( 3 );
+		const potFilesContent = potFiles.map( (file) => fs.readFileSync( file ) );
+		const xmlOutput = po2Android( potFilesContent );
```
Comment (fluiddot) [inline bin/po2android.js:126]: Now the `po2android` and `po2ios` scripts support multiple POT files, they basically merge all strings into a single object that is later on used for generating the localization strings files.

https://github.com/wordpress-mobile/gutenberg-mobile/blob/b42efc199d9dcc0bcd2f4269c915765aed08c148/bin/po2android.js#L87

File: src/block-experiments-setup.js [hunk 5]
```diff
@@ -1,8 +1,25 @@
+/**
+ * Internal dependencies
+ */
 // This file is to set up the jetpack/layout-grid block that currently lives in block-experiments/blocks/layout-grid
 import { registerBlock } from '../block-experiments/blocks/layout-grid/src';
+import setupLocale from './setup-locale';
+import { getTranslation as getLayoutGridTranslation } from './i18n-translations/layout-grid';
 
-export default function setupBlockExperiments( capabilities ) {
+const LAYOUT_GRID_LOCALE_DOMAIN = 'layout-grid';
```
Comment (fluiddot) [inline src/block-experiments-setup.js:9]: The domain for the Layout Grid block is `layout-grid`, [here](https://github.com/Automattic/block-experiments/blob/00dbb963f1615d9843f0d8a8479b585c7a0108e1/blocks/layout-grid/src/index.js#L45) you can see an example.

File: src/jetpack-editor-setup.js [hunk 6]
```diff
@@ -26,6 +28,8 @@ const supportedJetpackBlocks = {
 	},
 };
 
+const JETPACK_LOCALE_DOMAIN = 'jetpack';
```
Comment (fluiddot) [inline src/jetpack-editor-setup.js:31]: The domain for Jetpack is `jetpack`, [here](https://github.com/Automattic/jetpack/blob/677b2f563a278bb581576bc938d15cffe0589933/projects/plugins/jetpack/extensions/blocks/contact-info/address/edit.native.js#L37) you can see an example.

**Comment by mchowning** [comment]
I've got a bit of a task backlog right now, so I won't be able to get to this until next week. Could you take a look at this @jhnstn ?

**Review by jhnstn — APPROVED** [review]
The changes look good. 
I verified that none of the strings were changed via `genstrings` command. 
I did notice that 'ADD IMAGE' is not translated in the Jetpack  story block:


Screen Shot 2021-12-23 at 11 27 03 AM
I did find the string in `bundle/android/strings.xml` as is noted in the test steps

**Pull request merged**